core/server/middleware/error-handler.js

```diff
@@ -103,15 +71,68 @@ module.exports = function errorHandler(err, req, res, next) {
         'Cache-Control': 'no-cache, private, no-store, must-revalidate, max-stale=0, post-check=0, pre-check=0'
     });
 
-    // @TODO: does this resolves all use cases?
-    if (!req.headers.accept && !req.headers.hasOwnProperty('content-type')) {
-        req.headers.accept = 'text/html';
+    return err;
+};
+
+errorHandler.resourceNotFound = function resourceNotFound(req, res, next) {
+    // TODO, handle unknown resources & methods differently, so that we can also produce
+    // 405 Method Not Allowed
+    next(new errors.NotFoundError({message: i18n.t('errors.errors.resourceNotFound')}));
+};
+
+errorHandler.pageNotFound = function pageNotFound(req, res, next) {
+    next(new errors.NotFoundError({message: i18n.t('errors.errors.pageNotFound')}));
+};
+
+errorHandler.handleJSONResponse = function _handleJSONResponse(err, req, res, next) {
+    /*jshint unused:false */
+    // Make sure the error can be served
+    err = _private.prepareError(err, req, res);
+
+    // @TODO: jsonapi errors format ?
+    res.json({
+        errors: [{
+            message: err.message,
+            errorType: err.errorType,
+            errorDetails: err.errorDetails
+        }]
+    });
+};
+
+errorHandler.handleHTMLResponse = function handleHTMLResponse(err, req, res, next) {
+    /*jshint unused:false */
+    // Make sure the error can be served
+    err = _private.prepareError(err, req, res);
+
+    // @TODO reconsider this
```

Comment:
would be cool actually to not load available themes into config and rather ask a theme unit which knows all this stuff. then we could use this unit here. maybe we can extend the TODO a bit or add a task into theme audit issue?